### ISSUE TITLE
[Preview 4] DataGrid Crash in WinUI3 Desktop

### ISSUE THREAD

**Issue by eleanorleffler** [microsoft/microsoft-ui-xaml#4202]
**Describe the bug**

The CommunityToolKit DataGrid displays and is scrollable in UWP In WinUI3 Preview4 Desktop, the DataGrid crashes on load. Previously, in Preview3, the DataGrid would load but crash when made scrollable. See Screenshot#1 for current behavior.

**Steps to reproduce the bug**

1. Clone [WinUI3Preview4 Problems DataGrid repository](https://github.com/eleanorleffler/WinUI3Preview4_Problems_DataGrid).
2. Go to the DataGridWinUIPreview4 folder.
3. Open the DataGridWinUIPreview4 solution in Visual Studio 2019 Preview.
3. Build and run with Debug x64.
4. The application window will appear, and the application will crash.

**Expected behavior**

We expect the DataGrid to load and be scrollable when window is unable to display all of the data in the DataGrid - the same behavior we saw in UWP. See Screenshot#2

**Screenshots**

![CurrentBehavior](https://user-images.githubusercontent.com/9272634/108012426-f89e4480-6fce-11eb-87c3-1dff68353e37.PNG)

![CurrentBehavior2](https://user-images.githubusercontent.com/9272634/108012430-fb009e80-6fce-11eb-9425-8a7a6c20ae96.PNG)

Screenshot#1 - Current Behavior

![ExpectedBehavior](https://user-images.githubusercontent.com/9272634/108012442-0227ac80-6fcf-11eb-9b62-d64f1f7c273b.PNG)

Screenshot#2 - Expected Behavior

**Version Info**

NuGet package version: 

[Microsoft.Toolkit.Uwp.UI.Controls.DataGrid 8.0.0-preview4]
[Microsoft.WinUI 3.0.0-preview4.210210.4]

Targeting:
Target: Universal Windows
Target version: Windows 10, version 1809 (10.0; Build 17763)
Min version: Windows 10, version 1803 (10.0; Build 17134)

Windows app type:
| UWP              | Win32            |
| :--------------- | :--------------- |
| 		Yes 	   |  				  |

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              | 						 |
| May 2020 Update (19041)            | 						 |
| November 2019 Update (18363)       | 						 |
| May 2019 Update (18362)            | 						 |
| October 2018 Update (17763)        | 			Yes			 |
| April 2018 Update (17134)          | 						 |
| Fall Creators Update (16299)       | 						 |
| Creators Update (15063)            | 						 |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | 		Yes			|
| Xbox               | 					|
| Surface Hub        | 					|
| IoT                | 					|

**Additional context**

**Comment by StephenLPeters** [comment]
@RBrid and @michael-hawker FYI

**Comment by michael-hawker** [comment]
We've tested the DataGrid within the Toolkit sample app, so not sure what other factor is involved here. FYI @azchohfi.

**Comment by azchohfi** [comment]
Hello @eleanorleffler . Does this issue happens if the debugger is not attached to the application?

**Comment by eleanorleffler** [comment]
Hi @azchohfi . I have tried multiple combinations with the debugger, and the issues still occurs.

I also pulled down the Toolkit Sample app from GitHub and have a crash occur when I tried to "Launch the Windows Community Toolkit Sample App" - I see the app window appear but then go away.

**Comment by bkudiess** [comment]
Hi @eleanorleffler, could you try using the new Project Reunion 0.5 with the CommunityToolkit.WinUI.UI.Controls.DataGrid package?
I was able to make the same project work with these packages :) 

@azchohfi, I believe we should use the CommunityToolkit.WinUI.UI.Controls for WinUI3 desktop apps, is that right?

**Comment by azchohfi** [comment]
@bkudiess yes. We renamed all out packages from Microsoft.Toolkit.UWP -> CommunityToolkit.WinUI.

**Comment by eleanorleffler** [comment]
Hi @bkudiess, we were able to get it to work with Project Reunion 0.5! 

We had to follow the [Windows Community Toolkit for Project Reunion 0.5 documentation](https://devblogs.microsoft.com/ifdef-windows/windows-community-toolkit-for-project-reunion-0-5/) - which required us to change our "using" statements/namespaces and modify our project based on our target framework.

Thank you for following up with me!
